### ISSUE TITLE
Crash on same localized name for different entity codes

### ISSUE THREAD

**Issue by WickedSchnitzel** [DArkHekRoMaNT/PackingAnimals#4]
### Mod Version

2.6.1

### CommonLib Version

2.3.7

### Game Version

1.19.7

### OS

Windows

### Play mode

Multiplayer

### Problem

".. when attempting to click (left and right both) a single piece of rice grain into the inventory of an Iron Cage. The crash was instantaneous, and the rice was on the ground when I logged back in."

### Logs

```bash
Running on 64 bit Windows 10.0.19045.0 with 32701 MB RAM
Game Version: v1.19.7 (Stable)
5/8/2024 8:52:56 PM: Critical error occurred in the following mod: captureanimals@2.6.1
Loaded Mods: aculinaryartillery@1.1.4, ancientarmory@1.0.0, bettercrates@1.7.0, Ceramos@0.5.1, chiseltools@1.11.4, decor@1.0.15, extrachests@1.8.0, fantasycreatures@0.7.1, morecropsfix@1.0.6, fixedpaths@1.0.1, flourbags@1.0.2, foundrymod@1.0.0, giantgouramis@1.0.2, mannequinstand@1.0.3, millwright@1.1.5, moreicons_cs@1.1.0, peacockbasses@1.1.2, primitivesurvival@3.5.5, spyglass@0.5.1, Swordz@1.1.6, translocatorengineeringredux@1.5.0, truesunfishes@1.1.2, game@1.19.7, weaponpackalphaunoff@1.5.1, ancienttools@1.5.18, beehivekiln@1.5.3, betterfirepit@1.1.4, betterjonasdevices@1.1.0, betterruins@0.3.5, canclaims@1.0.3, canjewelry@0.2.29, carryon@1.7.4, commonlib@2.3.7, composter@1.1.0, expandedfoods@1.6.9, fairplayguardian@1.1.7, furniturelibrary@1.0.11, hangingbaskets@1.1.0, heatretention@1.0.2, herbarium@1.1.0, hudclock@3.4.0, sailboat@1.3.0, medievalexpansion@3.13.1, meteoricexpansion@1.2.5, morepiles@1.5.0, moreplaster@1.0.1, pelaguswinds@1.0.0, petai@2.2.4, rivers@2.4.0, simplestep@1.1.7, stonebakeoven@1.1.3, tentbag@1.1.2, th3dungeon@0.2.1-rc.1, thecritterpack@0.8.9, usefulstuff17@1.3.1, vinconomy@0.2.6, creative@1.19.7, vsinstruments@1.2.7, vsquest@1.2.0, vsserverpanel@0.1.4, survival@1.19.7, workbenchexpansion@1.8.0, xlib@0.8.3-pre.1, awearablelight@1.1.1, captureanimals@2.6.1, cob2@1.6.1, em@2.5.2, linearpower@0.2.1, outerspacetunesmaster@1.0.0, outerspacetunesvolume1@1.0.0, outerspacetunesvolume2@1.0.0, playercorpse@1.9.0, rifteye@0.3.6, stonequarry@3.2.1, vanity@2.1.2, wildcraftfruit@1.1.1, wolftaming@2.1.2, xskills@0.8.5-pre.1, bricklayers@2.5.2, cob@1.8.5, vsvillagecob@0.2.0
System.ArgumentException: An item with the same key has already been added. Key: Tundra bison (male)
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at CaptureAnimals.GuiDialogBait.UpdateText() in CaptureAnimals/Cage/GuiDialogBait.cs:line 86
   at CaptureAnimals.GuiDialogBait.<.ctor>b__1_0(Int32 n) in CaptureAnimals/Cage/GuiDialogBait.cs:line 18
   at Vintagestory.API.Common.InventoryBase.DidModifyItemSlot(ItemSlot slot, ItemStack extractedStack) in VintagestoryApi\Common\Inventory\InventoryBase.cs:line 480
   at Vintagestory.API.Common.ItemSlot.OnItemSlotModified(ItemStack sinkStack) in VintagestoryApi\Common\Inventory\ItemSlot.cs:line 432
   at Vintagestory.API.Common.ItemSlot.ActivateSlotLeftClick(ItemSlot sourceSlot, ItemStackMoveOperation& op) in VintagestoryApi\Common\Inventory\ItemSlot.cs:line 329
   at Vintagestory.API.Common.ItemSlot.ActivateSlot(ItemSlot sourceSlot, ItemStackMoveOperation& op) in VintagestoryApi\Common\Inventory\ItemSlot.cs:line 289
   at Vintagestory.API.Common.InventoryBase.ActivateSlot(Int32 slotId, ItemSlot sourceSlot, ItemStackMoveOperation& op) in VintagestoryApi\Common\Inventory\InventoryBase.cs:line 453
   at Vintagestory.API.Client.GuiElementItemSlotGridBase.SlotClick(ICoreClientAPI api, Int32 slotId, EnumMouseButton mouseButton, Boolean shiftPressed, Boolean ctrlPressed, Boolean altPressed) in VintagestoryApi\Client\UI\Elements\Impl\Interactive\Inventory\GuiElementItemSlotGridBase.cs:line 978
   at Vintagestory.API.Client.GuiElementItemSlotGridBase.OnMouseDownOnElement(ICoreClientAPI api, MouseEvent args) in VintagestoryApi\Client\UI\Elements\Impl\Interactive\Inventory\GuiElementItemSlotGridBase.cs:line 771
   at Vintagestory.API.Client.GuiComposer.OnMouseDown(MouseEvent mouseArgs) in VintagestoryApi\Client\UI\GuiComposer.cs:line 494
   at Vintagestory.API.Client.GuiDialog.OnMouseDown(MouseEvent args) in VintagestoryApi\Client\UI\Dialog\GuiDialog.cs:line 560
   at Vintagestory.Client.NoObf.GuiManager.OnMouseDown(MouseEvent args) in VintagestoryLib\Client\Systems\Gui\GuiManager.cs:line 390
   at Vintagestory.Client.NoObf.ClientMain.UpdateMouseButtonState(EnumMouseButton button, Boolean down) in VintagestoryLib\Client\ClientMain.cs:line 1909
   at Vintagestory.Client.HotkeyManager.TriggerHotKey(KeyEvent keyEventargs, IWorldAccessor world, IPlayer player, Boolean allowCharacterControls, Boolean isGlobal, Boolean fallBack, Boolean keyup) in VintagestoryLib\Client\HotkeyManager.cs:line 431
   at Vintagestory.Client.NoObf.ClientMain.OnMouseDownRaw(MouseEvent args) in VintagestoryLib\Client\ClientMain.cs:line 1836
   at Vintagestory.Client.NoObf.ClientPlatformWindows.Mouse_ButtonDown(MouseButtonEventArgs e) in VintagestoryLib\Client\ClientPlatform\Input.cs:line 198
   at OpenTK.Windowing.Desktop.NativeWindow.MouseButtonCallback(Window* window, MouseButton button, InputAction action, KeyModifiers mods)
--- End of stack trace from previous location ---
   at OpenTK.Windowing.Desktop.NativeWindow.RethrowCallbackExceptionsIfNeeded()
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 324
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass9_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 128
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93

Event Log entries for Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 5/6/2024 12:03:40 PM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.19.7.0, time stamp: 0x65310000
Faulting module name: openal32.dll, version: 1.23.0.0, time stamp: 0x63dd31ad
Exception code: 0x40000015
Fault offset: 0x00000000000df046
Faulting process id: 0x1788
Faulting application start time: 0x01da9fce9f0de9c3
Faulting application path: C:\Users\nacus\AppData\Roaming\Vintagestory\Vintagestory.exe
Faulting module path: C:\Users\nacus\AppData\Roaming\Vintagestory\Lib\openal32.dll
Report Id: 9d56c242-9452-48cd-b2d0-bd785a8a46ff
Faulting package full name: 
Faulting package-relative application ID:  }
--------------
{ TimeGenerated = 5/6/2024 12:00:58 PM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.19.7.0, time stamp: 0x65310000
Faulting module name: openal32.dll, version: 1.23.0.0, time stamp: 0x63dd31ad
Exception code: 0x40000015
Fault offset: 0x00000000000df046
Faulting process id: 0x3f50
Faulting application start time: 0x01da9fcdfbb326a5
Faulting application path: C:\Users\nacus\AppData\Roaming\Vintagestory\Vintagestory.exe
Faulting module path: C:\Users\nacus\AppData\Roaming\Vintagestory\Lib\openal32.dll
Report Id: 03378979-c308-44aa-bfc9-d71282c140d1
Faulting package full name: 
Faulting package-relative application ID:  }
--------------
{ TimeGenerated = 5/6/2024 11:56:15 AM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.19.7.0, time stamp: 0x65310000
Faulting module name: openal32.dll, version: 1.23.0.0, time stamp: 0x63dd31ad
Exception code: 0x40000015
Fault offset: 0x00000000000df046
Faulting process id: 0x48f4
Faulting application start time: 0x01da9fcdacd9bfa2
Faulting application path: C:\Users\nacus\AppData\Roaming\Vintagestory\Vintagestory.exe
Faulting module path: C:\Users\nacus\AppData\Roaming\Vintagestory\Lib\openal32.dll
Report Id: efa18a22-1f0a-4c42-8bf3-1e183d1b4b8b
Faulting package full name: 
Faulting package-relative application ID:  }
```

**Comment by DArkHekRoMaNT** [comment]
Did you edit/patch baits.json? Looks like you added rice grains to a bison twice

**Comment by WickedSchnitzel** [comment]
[addbaits.json](https://github.com/DArkHekRoMaNT/CaptureAnimals/files/15276331/addbaits.json)

We got
cob_tundrabison-female
cob_tundrabison-male
cob_tundrabison-male-hqm
tundrabison-stationary
on the server.

**Comment by DArkHekRoMaNT** [comment]
Hmm. Are `cob_tundrabison-male` and `cob_tundrabison-male-hqm` both named just "Tundra bison (male)"?

**Comment by WickedSchnitzel** [comment]
Yes, they are. But the name should not cause any bugs? 
We also have 2 language files in the pack, english and german. And both are in the "game" and "cob" folders, since some items need them to be either in game or cob.

**Comment by DArkHekRoMaNT** [comment]
Fixed in v2.7.0
